internal/xpkg/cache.go

```diff
@@ -17,95 +17,103 @@ limitations under the License.
 package xpkg
 
 import (
+	"compress/gzip"
 	"io"
 	"os"
 	"sync"
 
-	"github.com/google/go-containerregistry/pkg/name"
-	v1 "github.com/google/go-containerregistry/pkg/v1"
-	"github.com/google/go-containerregistry/pkg/v1/tarball"
 	"github.com/spf13/afero"
 
 	"github.com/crossplane/crossplane-runtime/pkg/errors"
 )
 
 const (
-	errGetNopCache = "cannot get an image from a NopCache"
+	errGetNopCache = "cannot get content from a NopCache"
 )
 
-// A Cache caches OCI images.
+const cacheContentExt = ".gz"
+
+// A Cache caches package content.
 type Cache interface {
-	Get(tag string, id string) (v1.Image, error)
-	Store(tag string, id string, img v1.Image) error
+	Has(id string) bool
+	Get(id string) (io.ReadCloser, error)
+	Store(id string, content io.ReadCloser) error
 	Delete(id string) error
 }
 
-// ImageCache stores and retrieves OCI images in a filesystem-backed cache in a
-// thread-safe manner.
-type ImageCache struct {
+// PackageCache stores and retrieves package content in a filesystem-backed
+// cache in a thread-safe manner.
+type PackageCache struct {
```

Comment:
Naming nit: Aren't all hypothetical instances of `Cache` (the interface) "a package cache"? What do you think about:

* `type Cache interface` -> `type PackageCache interface`
* `type PackageCache struct` -> `type FilesystemPackageCache struct` (or similar)

Reply:
Great call out -- I think I'll opt for `PackageCache` and `FsPackageCache` 👍🏻